cmd/rook/rook/rook.go

```diff
@@ -177,6 +180,24 @@ func GetOperatorImage(clientset kubernetes.Interface, containerName string) stri
 	return containerImage
 }
 
+func GetOperatorImagePullSecrets(clientset kubernetes.Interface) []v1.LocalObjectReference {
+
+	// If provided as a flag then use that value
+	if len(imagePullSecrets) > 0 {
+		var result []v1.LocalObjectReference
+		for _, name := range imagePullSecrets {
+			result = append(result, v1.LocalObjectReference{Name: name})
+		}
+		return result
+	}
+	// Getting the info of the operator pod
+
+	pod, err := k8sutil.GetRunningPod(clientset)
+	TerminateOnError(err, "failed to get pod")
```

Comment:
Add a more detailed error message. "failed to get operator pod details to determine image pull secrets"